### DISCUSSION TITLE
How do I generate a completely default maintenance function for a table

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#10235] [Help]
I have noticed that if I get the relationships right, the maintenance function I get "out of the box" in the **Data** part of Budibase is pretty much all I need.  How do I get to include that function in my published application without having to reinvent it all using the **Design** tool?     I am sure this must be possible but I cannot find out how.

**Comment by melohagan** [comment]
Hey @kpturner 

Have you tried creating an auto-generated screen? It gives you full CRUD functionality by using a built-in combination of side panels and the table block!

https://docs.budibase.com/docs/table-block#autogenerated-screen

**Reply by kpturner** [reply]
No but I will take a look

**Comment by peculiarism** [comment]
Hi @kpturner,

Budibase Product Manager here 👋🏻  I was wondering if you could provide more insight into the end state of your data, specifically:

- Did you use any filtering/views to reach your end state?
- Do you want end users to edit the data as it exists today?
- Was there any additional customisations made (queries/additional columns/bindings)?
